core/mantle_upgrade.go

```diff
@@ -13,50 +14,72 @@ var (
 		BVMETHMintUpgradeTime: u64Ptr(0),
 		MetaTxV2UpgradeTime:   u64Ptr(0),
 		MetaTxV3UpgradeTime:   nil, //TODO set upgrade timestamp
+		ProxyOwnerUpgradeTime: nil, //TODO set upgrade timestamp
 	}
 	MantleSepoliaUpgradeConfig = MantleUpgradeChainConfig{
 		ChainID:               params.MantleSepoliaChainId,
 		BaseFeeTime:           u64Ptr(1_704_891_600),
 		BVMETHMintUpgradeTime: u64Ptr(1_720_594_800),
 		MetaTxV2UpgradeTime:   u64Ptr(1_720_594_800),
 		MetaTxV3UpgradeTime:   u64Ptr(1_720_594_800),
+		ProxyOwnerUpgradeTime: u64Ptr(0),
 	}
 	MantleSepoliaQA3UpgradeConfig = MantleUpgradeChainConfig{
 		ChainID:               params.MantleSepoliaQA3ChainId,
 		BaseFeeTime:           u64Ptr(0),
 		BVMETHMintUpgradeTime: u64Ptr(0),
 		MetaTxV2UpgradeTime:   u64Ptr(0),
 		MetaTxV3UpgradeTime:   u64Ptr(1_717_689_600),
+		ProxyOwnerUpgradeTime: u64Ptr(0),
 	}
 	MantleSepoliaQA9UpgradeConfig = MantleUpgradeChainConfig{
 		ChainID:               params.MantleSepoliaQA9ChainId,
 		BaseFeeTime:           u64Ptr(0),
 		BVMETHMintUpgradeTime: u64Ptr(0),
 		MetaTxV2UpgradeTime:   u64Ptr(0),
 		MetaTxV3UpgradeTime:   u64Ptr(1_716_962_400),
+		ProxyOwnerUpgradeTime: u64Ptr(0),
 	}
 	MantleLocalUpgradeConfig = MantleUpgradeChainConfig{
 		ChainID:               params.MantleLocalChainId,
 		BaseFeeTime:           u64Ptr(0),
 		BVMETHMintUpgradeTime: u64Ptr(0),
 		MetaTxV2UpgradeTime:   u64Ptr(0),
 		MetaTxV3UpgradeTime:   u64Ptr(0),
+		ProxyOwnerUpgradeTime: u64Ptr(0),
 	}
 	MantleDefaultUpgradeConfig = MantleUpgradeChainConfig{
 		BaseFeeTime:           u64Ptr(0),
 		BVMETHMintUpgradeTime: u64Ptr(0),
 		MetaTxV2UpgradeTime:   u64Ptr(0),
 		MetaTxV3UpgradeTime:   nil,
+		ProxyOwnerUpgradeTime: nil,
 	}
 )
 
+// L2ProxyAdmin contract upgrade constants
+var (
+	// L2ProxyAdminAddress is the address of the L2ProxyAdmin contract
+	// predeploy
+	L2ProxyAdminAddress = common.HexToAddress("0x4200000000000000000000000000000000000018")
+
+	// OwnerSlot refers to the storage slot in the L2ProxyAdmin contract that
+	// holds the owner of the contract
+	OwnerSlot = common.BigToHash(big.NewInt(0))
+
+	// NewProxyAdminOwnerAddress is the address that the L2ProxyAdmin contract
+	// will be transferred to
+	NewProxyAdminOwnerAddress = common.HexToHash("0x09734bB3980906Bb217305EA6Bd34256feEAB105")
+)
+
```

Comment:
it's better to mv this to state_transition.go, only the upgrade height is stored here

Reply:
fixed